src/ariel-os-random/src/getrandom.rs

```diff
@@ -0,0 +1,17 @@
+//! Implementation of the getrandom ABI.
+//! Enable it by using the custom backend from getrandom. See [Opt-in Backends](https://github.com/rust-random/getrandom?tab=readme-ov-file#opt-in-backends)
+
+use crate::crypto_rng;
+use getrandom::Error;
+use rand_core::RngCore;
+
+#[unsafe(no_mangle)]
+unsafe extern "Rust" fn __getrandom_v03_custom(dest: *mut u8, len: usize) -> Result<(), Error> {
+    let buf = unsafe {
+        core::ptr::write_bytes(dest, 0, len);
```

Comment:
I doubt that this is really necessary (for every u8 value is inhabited, and if there are poison values in there, our try_fill_bytes will not read them), but as it's the upstream guidance, we better just follow it. (Also I'm not 100% sure that my understanding of poison values and &mut is right there).